client/tests/integration/transfer_asset.rs

```diff
@@ -10,7 +10,46 @@ use test_network::{Peer as TestPeer, *};
 use super::Configuration;
 
 #[test]
-fn client_can_transfer_asset_to_another_account() {
+fn simulate_transfer_quantity() {
+    simulate_transfer(200_u32, 20_u32, |id| AssetDefinition::quantity(id).build())
+}
+
+#[test]
+fn simulate_transfer_big_quantity() {
+    simulate_transfer(200_u128, 20_u128, |id| {
+        AssetDefinition::big_quantity(id).build()
+    })
+}
+
+#[test]
+fn simulate_transfer_fixed() {
+    simulate_transfer(
+        Fixed::try_from(200_f64).expect("Valid"),
+        Fixed::try_from(20_f64).expect("Valid"),
+        |id| AssetDefinition::fixed(id).build(),
+    )
+}
+
+#[should_panic]
+#[test]
+#[ignore = "long"]
+fn simulate_insufficient_funds() {
+    simulate_transfer(
+        Fixed::try_from(20_f64).expect("Valid"),
+        Fixed::try_from(200_f64).expect("Valid"),
+        |id| AssetDefinition::fixed(id).build(),
+    )
+}
+
+// TODO add tests when the transfer uses the wrong AssetId.
+
+fn simulate_transfer<T: Into<AssetValue> + Clone, D: FnOnce(AssetDefinitionId) -> AssetDefinition>(
+    starting_amount: T,
+    amount_to_transfer: T,
+    value_type: D,
+) where
+    Value: From<T>,
```

Comment:
Didn't know we can add trait bounds to a type not from a function declaration...